### ISSUE TITLE
evaluate results

### ISSUE THREAD

**Issue by jannaescur** [vrenkens/nabu#12]
Hi again!

I would like to evaluate the results after training the LAS model with TIMIT corpus.
1. Is there any way to get the file with the transcriptions generated to be compared with the groundtruth?
2. Does it make sense to calculate the WER if we are working with phonemes?

thanks

**Comment by vrenkens** [comment]
1. Yes you should do `run decode ...` instead of `run test .... `. You will then get the output hypotheses in the expdir.
2. We're actually calculating CER instead of WER. If you want to use WER instead you can try ro convert the phoneme sequences to text with some dictionaries or something (not optimal), but this  is none trivial.

**Comment by jannaescur** [comment]
great! and is there any script to calculate CER?

**Comment by vrenkens** [comment]
if you run the test script that't what it's calculating

**Comment by jannaescur** [comment]
mmh, but running test I get the loss..

**Comment by vrenkens** [comment]
It calls the CER a loss since it doesn't know what its evaluating. If your test_evaluator is the decoder evaluator the loss it gives is the CER

**Comment by jannaescur** [comment]
perfect! thanks

**Comment by MartaCostaJussa** [comment]
is a CER of 81% what you obtain in your tests? it is a really high error, isn't it?

**Comment by vrenkens** [comment]
No :p that is indeed a very high error rate. Can you give me some more specifics about the experiment?

**Comment by jannaescur** [comment]
i am getting now a CER of 39%. It is still high but I cannot find papers that compute the CER with the TIMIT dataset to compare, so I dont know if it is still toooo high. 
I just run your code without changing any parameter using TIMIT (with the small test set)

**Comment by vrenkens** [comment]
That is still too high.. When I run the recipe I get a CER around 24%.. This is not state-of-the-art performance. They do all sorts of tricks to get it to work better, but I don't have the time to implement those. It's strange that your performance is so much worse if you just run the recipe.. How Could you give me some plots of the training and validation loss during training?

**Comment by jannaescur** [comment]
![training_loss-iloveimg-cropped](https://user-images.githubusercontent.com/22589206/35217069-2db8d334-ff6a-11e7-8c4f-45cb3f13a9ba.jpg)
![validation_loss-iloveimg-cropped](https://user-images.githubusercontent.com/22589206/35217070-2dd7a6b0-ff6a-11e7-81a5-89bd1d741932.jpg)

**Comment by vrenkens** [comment]
It seems to me that it has not finished training... Could you give the training config that you are using?

**Comment by jannaescur** [comment]
it is the standard trainer with the parameters of your previous commit (initial learning rate: 5e-3, batch size: 128...)

**Comment by vrenkens** [comment]
Hmm, it seems I modified the recipe by accident. I will fix it and get back to you.
